providers/softlayer/src/test/java/org/jclouds/softlayer/features/NetworkApiMockTest.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.softlayer.features;
+
+import com.google.common.collect.ImmutableList;
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+import org.jclouds.http.config.JavaUrlHttpCommandExecutorServiceModule;
+import org.jclouds.softlayer.SoftLayerApi;
+import org.jclouds.softlayer.config.SoftLayerParserModule;
+import org.jclouds.softlayer.domain.Network;
+import org.jclouds.softlayer.domain.Subnet;
+import org.jclouds.softlayer.internal.BaseSoftLayerMockTest;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.google.common.collect.Iterables.size;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+/**
+ * Mock tests for the {@link NetworkApi} class.
+ */
+@Test(groups = "unit", testName = "SecurityGroupApiMockTest")
+public class NetworkApiMockTest extends BaseSoftLayerMockTest {
```

Comment:
great start @swaqos!

Similarly to other provider, jclouds also tests fallbacks strategies when the operations are not successful.
For example, `listNetworks` is annotated with `@Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)` so a test similar to https://github.com/jclouds/jclouds/blob/master/providers/packet/src/test/java/org/jclouds/packet/features/FacilityApiMockTest.java#L45 is required.

Please add the missing "bad case" tests for the operations in this MockTest class to improve the coverage